prost-build/src/code_generator.rs

```diff
@@ -670,6 +679,10 @@ impl<'a> CodeGenerator<'a> {
         self.depth += 1;
 
         for variant in variant_mappings.iter() {
+            if variant.deprecated {
+                self.push_indent();
+                self.buf.push_str("#[allow(deprecated)]\n");
```

Comment:
Why is the `allow` statement now needed and not is previous version?

Reply:
This direct call will raise deprecated warning for every call of `as_str_name`, so we need a `#[allow(deprecated)]` for normal call.
For deprecated enum, direct call `as_str_name` will still raise warning.